components/camel-ai/camel-langchain4j-tools/src/main/java/org/apache/camel/component/langchain4j/tools/LangChain4jToolsEndpoint.java

```diff
@@ -0,0 +1,198 @@
+
+    @Metadata(label = "consumer")
+    @UriParam(description = "List of Tool parameters in the form of parameter.<name>=<type>", prefix = "parameter.",
+              multiValue = true, enums = "string,integer,number,object,array,boolean,null")
+    private Map<String, String> parameters;
+
+    @Metadata(label = "consumer,advanced")
+    @UriParam(description = "Tool's Camel Parameters, programmatically define Tool description and parameters")
+    private CamelSimpleToolParameter camelToolParameter;
+
+    public LangChain4jToolsEndpoint(String uri, LangChain4jToolsComponent component, String toolId, String tags,
+                                    LangChain4jToolsConfiguration configuration) {
+        super(uri, component);
+        this.toolId = toolId;
+        this.tags = tags;
+        this.configuration = configuration;
+    }
+
+    @Override
+    public Producer createProducer() throws Exception {
+        return new LangChain4jToolsProducer(this);
+    }
+
+    @Override
+    public Consumer createConsumer(Processor processor) throws Exception {
+        ToolSpecification.Builder toolSpecificationBuilder = ToolSpecification.builder();
+
+        if (camelToolParameter != null) {
+            toolSpecificationBuilder.description(camelToolParameter.getDescription());
+
+            for (NamedJsonSchemaProperty namedJsonSchemaProperty : camelToolParameter.getProperties()) {
+                toolSpecificationBuilder.addParameter(namedJsonSchemaProperty.getName(),
+                        namedJsonSchemaProperty.getProperties());
+            }
+        } else if (description != null) {
+            toolSpecificationBuilder.description(description);
+
+            if (parameters != null) {
+                parameters.forEach((name, type) -> toolSpecificationBuilder.addParameter(name, JsonSchemaProperty.type(type)));
+            }
+        } else {
+            // Consumer without toolParameter or description
+            throw new IllegalArgumentException(
+                    "In order to use the langchain4j component as a consumer, you need to specify at least description, or a camelToolParameter");
+        }
+
+        String simpleDescription = null;
+        if (description != null) {
+            simpleDescription = StringHelper.dashToCamelCase(description.replace(" ", "-"));
+        }
+
+        ToolSpecification toolSpecification = toolSpecificationBuilder
+                .name(simpleDescription)
+                .build();
+
+        CamelToolSpecification camelToolSpecification
+                = new CamelToolSpecification(toolSpecification, new LangChain4jToolsConsumer(this, processor));
+        final CamelToolExecutorCache executorCache = CamelToolExecutorCache.getInstance();
+
+        String[] splitTags = TagsHelper.splitTags(tags);
+        for (String tag : splitTags) {
+            executorCache.put(tag, camelToolSpecification);
+        }
+
+        return camelToolSpecification.getConsumer();
```

Comment:
When creating a consumer, you also need to call `configureConsumer(consumer)` before returning the object.

Reply:
Ahh, good one. BTW, I've just noticed that we have the same problem on the `camel-langchain4j-chat` component. I'll fix here first and later there. Thanks!